controller/sync.go

```diff
@@ -249,20 +242,28 @@ func (m *appStateManager) SyncAppState(app *v1alpha1.Application, state *v1alpha
 				m.isSelfReferencedObj(live, appLabelKey, trackingMethod)
 		}),
 		sync.WithManifestValidation(!syncOp.SyncOptions.HasOption(common.SyncOptionsDisableValidation)),
-		sync.WithNamespaceCreation(syncOp.SyncOptions.HasOption("CreateNamespace=true"), func(un *unstructured.Unstructured) bool {
-			if un != nil && kube.GetAppInstanceLabel(un, cdcommon.LabelKeyAppInstance) != "" {
-				kube.UnsetLabel(un, cdcommon.LabelKeyAppInstance)
-				return true
```

Comment:
It seems that this logic isn't present in the `syncNamespace` function. Is there a reason to stop cleaning the instance label from the namespace?

Reply:
Don't think there's any particular reason; think this was an artifact from when the resource tracking was always being set. I've re-added it, but what is the reason for having this? If this is significant, I'd add a test to check that this gets unset.